indico/modules/rb_new/client/js/common/rooms/RoomEditModal.jsx

```diff
@@ -0,0 +1,419 @@
+    //     console.warn(acceptedFiles);
+    //     console.warn(rejectedFiles);
+    // };
+
+    onEditHours = () => {
+    };
+
+    onEditAttributes = () => {
+        this.addAttributes();
+    };
+
+    async fetchDetailedRoom() {
+        const {roomId} = this.props;
+        let response;
+        try {
+            response = await indicoAxios.get(fetchRoomURL({room_id: roomId}));
+        } catch (error) {
+            handleAxiosError(error);
+            return;
+        }
+        this.setState({room: camelizeKeys(response.data)});
+    }
+
+    async fetchRoomAttributes() {
+        const {roomId} = this.props;
+        let response;
+        try {
+            response = await indicoAxios.get(fetchRoomAttributesURL({room_id: roomId}));
+        } catch (error) {
+            handleAxiosError(error);
+            return;
+        }
+        this.setState({roomAttributes: response.data});
+    }
+
+    async fetchAttributes() {
+        let response;
+        try {
+            response = await indicoAxios.get(fetchAttributesURL());
+        } catch (error) {
+            handleAxiosError(error);
+            return;
+        }
+        this.setState({attributes: response.data});
+    }
+
+    handleCloseModal = () => {
+        const {onClose} = this.props;
+        onClose();
+    };
+
+    handleSubmit = async (formData) => {
+        const {actions: {updateRoom}, room: {id}} = this.props;
+        const rv = await updateRoom(id, formData);
+        if (rv.error) {
+            return rv.error;
+        }
+    };
+
+    renderImage = (position) => {
+        return (
+            <>
+                {/* <Dropzone key={position} onDrop={this.onDrop}> */}
+                <SpriteImage pos={position} />
+                {/* </Dropzone> */}
+            </>);
+    };
+
+    removeAttribute = (attribute) => {
+        const {roomAttributes} = this.state;
+        this.setState({roomAttributes: _.omit(roomAttributes, attribute)});
+    };
+
+    renderAttributes = () => {
+        const {attributes, roomAttributes} = this.state;
+        if (!attributes) {
+            return;
+        }
+        return attributes.map(attribute => (
+            <>
+                {_.has(roomAttributes, attribute.name) && (
+                    <Field name={attribute.name}
+                           component={ReduxFormField}
+                           label={attribute.name}
+                           as={Input}
+                           icon={{name: 'remove', color: 'red', circular: true, link: true, onClick: () => this.removeAttribute(attribute.name)}} />
+                )} </>
+
+        ));
+    };
+
+    renderAddButton = (type) => {
+        const {attributes} = this.state;
+        return (
+            <Button disabled={attributes.length === 0}
+                    default
+                    size="tiny"
+                    floated="right"
+                    onClick={type === 'attributes' ? this.onEditAttributes : this.onEditHours}>
```

Comment:
i think usually the function passed to `onSomething` should be called `handleXXX`, e.g. `handleEditAttributesClick`